src/Microsoft.Azure.Batch.SoftwareEntitlement/EntitlementVerifier.cs

```diff
@@ -55,11 +60,38 @@ public EntitlementVerifier(NodeEntitlementReader entitlementReader)
                 return Errorable.Failure<NodeEntitlements>($"Token does not grant entitlement for {request.IpAddress}");
             }
 
+            // There should always be a CPU core count in the token, even though the
+            // particular version of the API we're using may not require it to be
+            // checked.
+            if (!entitlement.CpuCoreCount.HasValue)
+            {
+                return Errorable.Failure<NodeEntitlements>("Token does not grant entitlement for any CPU cores");
+            }
+
+            // The presence of a CPU core count in the request indicates that we need to validate
+            // it against the maximum permitted number in the entitlement.
+            // Checking whether we're *supposed* to have a value provided in the request is
+            // beyond the scope of this method.
+            if (request.CpuCoreCount.HasValue &&
+                request.CpuCoreCount.Value > entitlement.CpuCoreCount.Value)
```

Comment:
Only the `Server` code needs to know about API versions at the moment. If an API version mandates the presence of CPU core count on the request, the `Server` layer is responsible for initializing the `request` object accordingly, or returning an error to the user if it's not provided.

So if the CPU count is not specified on the request, that can be interpreted as "we don't need to check this".

The `entitlement` object, on the other hand, is expected to _always_ contain a CPU core count, since it is derived from the token.